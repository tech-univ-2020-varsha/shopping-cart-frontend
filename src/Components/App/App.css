.App {
  flex-grow: 1;
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  font-family: 'Poppins'
}

