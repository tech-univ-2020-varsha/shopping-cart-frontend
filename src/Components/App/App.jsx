import React from 'react';
import './App.css';
import Header from '../Header/header';
import useProduct from '../../hooks/useProduct';

const App = () => {
  const [products, filterProduct, callComplete] = useProduct();
  if (callComplete) { console.log(filterProduct); }
  return (
    <div className="App">
      <Header />
    </div>
  );
};


export default App;
