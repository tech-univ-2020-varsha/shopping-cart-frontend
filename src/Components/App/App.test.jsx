import React from 'react';
import { render } from '@testing-library/react';
import axios from 'axios';
import App from './App';
import URL from '../../constants/url';

describe('The App component', () => {
  const mockAxios = jest.spyOn(axios, 'get');
  mockAxios.mockResolvedValue({ data: [] });

  it('should render correctly', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should make an api call to fetch product details', () => {
    const { asFragment } = render(<App />);
    expect(mockAxios).toHaveBeenCalledWith(`${URL}/cart`);
  });
});
