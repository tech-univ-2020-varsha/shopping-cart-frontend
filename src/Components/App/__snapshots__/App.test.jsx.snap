// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The App component should render correctly 1`] = `
<DocumentFragment>
  <div>
    Loading!...
  </div>
</DocumentFragment>
`;
